src/components/FormValidator.js

```diff
@@ -34,12 +34,23 @@ class FormValidator {
     errorElement.textContent = "";
   }
 
-  toggleInputError(inputElement) {
-    if (!inputElement.validity.valid) {
-      this._showInputError(inputElement);
+  toggleInputError() {
+    const hasInvlidInputs = this._hasInvalidInput();
+
+    if (hasInvlidInputs) {
+      const invalidInputsElement = this._inputList.filter((inputElement) => {
+        return !inputElement.validity.valid;
+      });
+
+      invalidInputsElement.forEach((inputElement) => {
+        this._showInputError(inputElement);
+      });
+
       this.disableButton();
```

Comment:
`  this.disableButton();`


Please, delete `this.disableButton();` and ` this._enableButton();` from `toggleInputError` because you can't control the button depending only on **one** input. According to the theory you had method `_toggleButtonState`  with this code


```js

  _toggleButtonState() {
    if (this._hasInvalidInput()) {
      this.disableButton();
    } else {
      this._enableButton();
    }
  }
```


You need to call it inside `input` handler


```js
    this._inputList.forEach((inputElement) => {
      inputElement.addEventListener("input", () => {
        this.toggleInputError(inputElement);
        this._toggleButtonState();
      });
    });

```


And now everything will work fine


So, bring the code of `toggleInputError` back 


```js

  toggleInputError(inputElement) {
    if (!inputElement.validity.valid) {
      this._showInputError(inputElement);
    } else {
      this._hideInputError(inputElement);
    }
  }

```